analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/IsNull.cs

```diff
@@ -30,5 +30,11 @@ internal static class IsNull
             && value.HasConstraint<ObjectConstraint>()
                 ? context.SetOperationConstraint(BoolConstraint.From(value.HasConstraint(ObjectConstraint.Null)))
                 : context.State;
+
+        public static ProgramState LearnBranchingConstraint(ProgramState state, IIsNullOperationWrapper isNull, bool useOpposite) =>
```

Comment:
I find "useOpposite" a bit confusing here. In this case, I would call it `notNullBranch` or something like that. In the SyntaxTree it is called "WhenNotNull".

Reply:
It's not perfect, but it follows the `useOpposite` from top level. So it indicates the fact that it's the flip/flop in negation chain.